clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -389,6 +389,19 @@ public void testMetricsReporterAutoGeneratedClientId(GroupProtocol groupProtocol
         consumer.close(CloseOptions.timeout(Duration.ZERO));
     }
 
+    @ParameterizedTest
+    @EnumSource(GroupProtocol.class)
+    public void testMetricsRemovedOnClose(GroupProtocol groupProtocol) {
+        Properties props = new Properties();
+        props.setProperty(ConsumerConfig.GROUP_PROTOCOL_CONFIG, groupProtocol.name());
+        props.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+        consumer = newConsumer(props, new StringDeserializer(), new StringDeserializer());
+
+        assertTrue(consumer.metrics().size() > 1, "The consumer should have created many metrics");
+        consumer.close(CloseOptions.timeout(Duration.ZERO));
+        assertTrue(consumer.metrics().size() <= 1, "The consumer should have removed all of its metrics");
```

Comment:
Curious about the `> 1` and `<= 1`, reads like there's usually a metric just hanging around in there?